### PR TITLE
New test - rpm-ostree-container-bootc + related improvements

### PR DESCRIPTION
**Pull request by jikortus**
The test is based on previous [work](https://github.com/rhinstaller/kickstart-tests/pull/1037) from Vladimir Slavik.

### CODE REVIEW

**Comment by jikortus** [comment]
It seems that the test was failing with RHEL-9 due to `KSTEST_OS_NAME` and `KSTEST_OS_VERSION` environment variables extracted from boot.iso not being passed to the test environment. I added another commit that should hopefully fix it (works OK in my setup).

File: scripts/run_kickstart_tests.sh [hunk 1]
```diff
@@ -176,8 +176,10 @@ if [[ $? -ne 0 ]]; then
 fi
 ISO_OS_NAME=$(echo "${output}" | grep 'NAME=')
 ISO_OS_NAME="${ISO_OS_NAME##NAME=}"
+export KSTEST_OS_NAME="${ISO_OS_NAME}"
```
Comment (rvykydal) [inline scripts/run_kickstart_tests.sh:179]: This should not be needed as in ks.in files we use @KSTEST_OS_NAME@ and @KSTEST_OS_VERSION@ macros as in https://github.com/rhinstaller/kickstart-tests/blob/208da7737a2662663634855cfa005ae8ccdd8b77/repo-install.ks.in#L20 (they are substituted with the values 5 lines below). (for the reference, documented here: https://github.com/rhinstaller/kickstart-tests/blob/master/README.rst#chapter-4-environment-variables)

File: rpm-ostree-container-bootc.sh [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+# Ignore unused variable parsed out by tooling scripts as test tags metadata
+# shellcheck disable=SC2034
+TESTTYPE="payload ostree bootc reboot skip-on-rhel-8 skip-on-rhel-10"
+# skip-on-rhel"
+
+. ${KSTESTDIR}/functions.sh
+
+os_major=${KSTEST_OS_VERSION%%.*}
+if [ "${KSTEST_OS_NAME}" == "rhel" ]; then
```
Comment (rvykydal) [inline rpm-ostree-container-bootc.sh:27]: @KSTEST_OS_NAME@ and @KSTEST_OS_VERSION@ should be used in .ks.in (https://github.com/rhinstaller/kickstart-tests/blob/master/README.rst#chapter-4-environment-variables)

We updated similar stuff recently and use the variables like this:  https://github.com/rhinstaller/kickstart-tests/blob/208da7737a2662663634855cfa005ae8ccdd8b77/repo-install.ks.in#L20 which is not ideal. I'd maybe add an rhel_os_major_version function to the https://github.com/rhinstaller/kickstart-tests/blob/master/scripts-lib.sh that would replace is_rhel8 and is_rhel9. But that is something  I'd do independently on top of your PR.

**Comment by rvykydal** [comment]
@jikortus thank you for the PR! I will look at it more, for now I have just a clarification about using KSTEST_ variables in .ks.in files.
Also, could we move the first two commits (IIUIC they are not required by the test itself) to a separate PR? I believe tt would be easier to get the changes reviewed and included in separate steps.

**Comment by rvykydal** [comment]
Sorry, I started to look at the patchset, did some wrong review and deleted it.
Thank you for the PR!

**Comment by rvykydal** [comment]
Nice work. I am suggesting updates upon your great work in https://github.com/rhinstaller/kickstart-tests/pull/1116. Mostly because our common handling of os (rhel / fedora) substitutions at various places is not very clear at first sight:

We do all the per-os substitutions in the .ks.in. I guess you did them also in .sh because you were not aware of the fragments mechanism we use for kickstart commands (`%ksappend`). For %post scripts we use `@KSTESTS_OS_NAME` etc.: https://github.com/rhinstaller/kickstart-tests/pull/1116/commits/41045cf2ce7167c6d2b5581816eed4a655b35c59

We should even do it more concisely as in the follow-up commit of the PR: https://github.com/rhinstaller/kickstart-tests/pull/1116/commits/f965f8bda9d98d09098c5db34d495b4eb0e5803e

I would also suggest submitting the first commit of this PR as a separate PR as it is not required by the test and reviewing it separately will simplify the process of getting the changes in (I would maybe even split it to 2 PRs).

File: rpm-ostree-container-bootc.ks.in [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+# Validate on the first boot.
+%ksappend validation/success_on_first_boot.ks
+
+%post
+# Checks in %post
+# Check of the remote URL is appended by the test script
+
+# Checks after boot
+cat >> /opt/kickstart-tests/kickstart-test.sh << 'EOF'
```
Comment (rvykydal) [inline rpm-ostree-container-bootc.ks.in:21]: This part was failing for me with file not found, updated here: https://github.com/rhinstaller/kickstart-tests/pull/1116/commits/41045cf2ce7167c6d2b5581816eed4a655b35c59

**Comment by jikortus** [comment]
@rvykydal, thanks a lot for your feedback and for the improvements you've made in #1116. At this point, I think we could abandon this PR in favour of yours. I'm mostly aware of how the fragments work and I like introducing the new 'variable' for the container image, I didn't really think of approaching it this way :-).

I'll create a new PR for passing the variables (5da51cc) to make it more granular and easier to review. BTW, should I drop 56f198b altogether? It wouldn't likely be needed anymore with the new way to pass the container image URL, OTOH the two variables aren't (IIUIC) passed to the container test environment when executing the tests via `run_kickstart_tests.sh` - not sure if/how wrong that would be in the end.

**Comment by rvykydal** [comment]
> @rvykydal, thanks a lot for your feedback and for the improvements you've made in #1116. At this point, I think we could abandon this PR in favour of yours. I'm mostly aware of how the fragments work and I like introducing the new 'variable' for the container image, I didn't really think of approaching it this way :-).

Okay, let's continue with https://github.com/rhinstaller/kickstart-tests/pull/1116, feel free to add a review if you want.

> 
> I'll create a new PR for passing the variables ([5da51cc](https://github.com/rhinstaller/kickstart-tests/commit/5da51cccc5cdecd67e9abef5830fa586c038c651)) to make it more granular and easier to review. 

Thank you.

> BTW, should I drop [56f198b](https://github.com/rhinstaller/kickstart-tests/commit/56f198b9a2ea51660a7b345f1e2e3c6ee9210463) altogether? It wouldn't likely be needed anymore with the new way to pass the container image URL, OTOH the two variables aren't (IIUIC) passed to the container test environment when executing the tests via `run_kickstart_tests.sh` - not sure if/how wrong that would be in the end.

Yes, I'd drop it at this point.

Closing as obsoleted by  https://github.com/rhinstaller/kickstart-tests/pull/1116 and https://github.com/rhinstaller/kickstart-tests/pull/1117

**Pull request closed without merging**